pkg/cli/cli.go

```diff
@@ -0,0 +1,107 @@
+package cli
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/cyberark/conjur-oss-suite-release/pkg/changelog"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/github"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/http"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/repositories"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/template"
+)
+
+type Options struct {
+	APIToken           string
+	Date               time.Time
+	OutputFilename     string
+	OutputType         string
+	RepositoryFilename string
+	Version            string
+}
+
+func RunParser(options Options) {
+	templateInfo, err := template.GetTemplateInfo(options.OutputType)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	options.setOutputFilename(templateInfo.OutputFileName)
+
+	log.Printf("Parsing linked repositories...")
+	repoConfig, err := repositories.NewConfig(options.RepositoryFilename)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	if options.OutputType == "unreleased" {
+		// This is an in-place operation
+		repoConfig.SelectUnreleased()
+	}
```

Comment:
Fly by comment: 

Never write mutating APIs unless you're forced to for performance reasons.  Let `SelectUnreleased` return a new object instead.

Reply:
Hey Jonah! Your feedback is more than welcome here, but I should point out that this package is primarily a restructuring, and this code was written much earlier. If there's an important change, it might be valuable to bring up in slack or make an issue so it doesn't get lost. I think it's a bit outside of the scope of this story though.